packages/js/components/src/form/form.tsx

```diff
@@ -163,8 +163,10 @@ function FormComponent< Values extends Record< string, any > >(
 	);
 
 	useEffect( () => {
-		validate( values );
-	}, [] );
+		initialValues.current = props.initialValues ?? ( {} as Values );
+		setValuesInternal( initialValues.current );
```

Comment:
I think this could lead to unexpected behavior.  If updates were made to a form and then the `initialValues` were subsequently updated, any dirty state would be lost.

Reply:
It's true, but does make sense to have dirty values if the initialValues of the form contains changes?

Reply:
I also think we should reset the `dirtyState` in that case, which will mean resetting the `touched` state as well I think